src/components/PostDetails.tsx

```diff
@@ -1,117 +1,147 @@
+  const [addNewComment, setAddNewComment] = useState(false);
+
+  useEffect(() => {
+    setLoading(true);
+    if (commentsLoadError) {
+      setCommentsLoadError('');
+    }
+
+    if (comments) {
+      setComments(null);
+    }
+
+    if (addNewComment) {
+      setAddNewComment(false);
+    }
+
+    client.get<Comment[]>(`/comments?postId=${id}`)
+      .then(setComments)
+      .catch(() => setCommentsLoadError("Can't load comments for current post"))
+      .finally(() => setLoading(false));
+  }, [post]);
+
+  function handleDeleteComment(commentId: number) {
+    client.delete(`/comments/${commentId}`)
+      .then(() => {
+        setComments(
+          current => current?.filter(item => item.id !== commentId) || null,
+        );
+        setCommentsDelError([]);
+      })
+      .catch(() => setCommentsDelError(current => [...current, commentId]));
+  }
+
+  const handleAddNewComment = (newComment: Comment) => {
+    setComments(current => (current ? [...current, newComment] : [newComment]));
+  };
 
-export const PostDetails: React.FC = () => {
   return (
     <div className="content" data-cy="PostDetails">
       <div className="content" data-cy="PostDetails">
         <div className="block">
           <h2 data-cy="PostTitle">
-            #18: voluptate et itaque vero tempora molestiae
+            {`#${id}: ${title}`}
           </h2>
 
           <p data-cy="PostBody">
-            eveniet quo quis
-            laborum totam consequatur non dolor
-            ut et est repudiandae
-            est voluptatem vel debitis et magnam
+            {body}
           </p>
         </div>
 
-        <div className="block">
+        {loading && (
           <Loader />
+        )}
 
-          <div className="notification is-danger" data-cy="CommentsError">
-            Something went wrong
-          </div>
+        {!loading && (
+          <div className="block">
+            {commentsLoadError && (
+              <div
+                className="notification is-danger"
+                data-cy="PostsLoadingError"
+              >
+                {commentsLoadError}
+              </div>
+            )}
 
-          <p className="title is-4" data-cy="NoCommentsMessage">
-            No comments yet
-          </p>
+            {comments && (!comments.length ? (
+              <p className="title is-4" data-cy="NoCommentsMessage">
+                No comments yet
+              </p>
+            ) : (
+              <>
+                <p className="title is-4">Comments:</p>
 
-          <p className="title is-4">Comments:</p>
+                {comments.map(comment => (
+                  <article
+                    className="message is-small"
+                    data-cy="Comment"
+                    key={comment.id}
+                  >
+                    <div className="message-header">
+                      <a href={`mailto:${comment.email}`} data-cy="CommentAuthor">
+                        {comment.name}
```

Comment:
Don't forget about the destructuring

Reply:
I think that destructuring is not needed here, at the top of the code there is an object with which there will be a conflict

Reply:
you dont need whole obj here, so you can easily to make it here